taca/deliver/deliver.py

```diff
@@ -0,0 +1,385 @@
+            except AttributeError as e:
+                raise DelivererError(
+                    "the path '{}' could not be expanded - reason: {}".format(
+                        path,e))
+    
+class ProjectDeliverer(Deliverer):
+    
+    def __init__(self, projectid=None, sampleid=None, **kwargs):
+        super(ProjectDeliverer,self).__init__(
+            projectid,
+            sampleid,
+            **kwargs)
+    
+    def deliver_project(self):
+        """ Deliver all samples in a project to the destination specified by 
+            deliverypath
+            
+            :returns: True if all samples were delivered successfully, False if
+                any sample was not properly delivered or ready to be delivered
+        """
+        self.log.info(
+            "Delivering {} to {}".format(str(self),self.deliverypath))
+        try:
+            sampleentries = self.project_sample_entries()
+        except DelivererDatabaseError as e:
+            self.log.error(
+                "error '{}' occurred during delivery of {}".format(
+                    str(e),str(self)))
+            raise
+        # right now, don't catch any errors since we're assuming any thrown 
+        # errors needs to be handled by manual intervention
+        status = True
+        print(", ".join([s.get('sampleid') for s in sampleentries.get('samples')]))
+        for sampleentry in sampleentries.get('samples',[]):
+            st = SampleDeliverer(
+                self.projectid,sampleentry.get('sampleid')
+            ).deliver_sample(sampleentry)
+            status = (status and st)
+            
+        if status:
+            self.update_delivery_status()
+        return status
+
+class SampleDeliverer(Deliverer):
+    """
+        A class for handling sample deliveries
+    """
+    def __init__(self, projectid=None, sampleid=None, **kwargs):
+        super(SampleDeliverer,self).__init__(
+            projectid,
+            sampleid,
+            **kwargs)
+        
+    def deliver_sample(self, sampleentry=None):
+        """ Deliver a sample to the destination specified by the config.
+            Will check if the sample has already been delivered and should not 
+            be delivered again or if the sample is not yet ready to be delivered.
+            
+            :params sampleentry: a database sample entry to use for delivery
+                but not sent to the receiver
+            :returns: True if sample was successfully delivered or was previously 
+                delivered, False if sample was not yet ready to be delivered
+            :raises DelivererDatabaseError: if an entry corresponding to this
+                sample could not be found in the database
+            :raises DelivererReplaceError: if a previous delivery of this sample
+                has taken place but should be replaced
+            :raises DelivererError: if the delivery failed
```

Comment:
not sure if that would be something desired or useful, but just crossed my mind: maybe an option to force the delivery? if that could ever be needed